Automattic/pocket-casts-ios
# Patron: update upgrade screen

Updates the new upgrade screen to display prices right away and allow the purchase without the need to show the modal.

## To test

### Pre-testing setup:
1. Open the Xcode Project
3. Hit <kbd>Command</kbd>+<kbd>Shift</kbd>+<kbd>,</kbd> to open the Scheme editor
4. Click the options tab
5. Select the `Pocket Casts Configuration.storekit` file in the StoreKit configuration dropdown

### Test the current flow

1. Enable the `patron` flag in `FeatureFlag.swift`
2. Do a clean install of the app
3. Go through the whole onboarding process and buy any plan
4. ✅ Make sure you see the confetti screen at the end
5. Go to Settings > Developer > tap "Set to No Plus"
6. Go to Podcasts > tap the Folder button
7. Buy any plan
8. ✅ Make sure you see the confetti screen
9. On Xcode, go to Debug > StoreKit > Manage Transactions
10. Delete Everything
11. Display the `.sorekit` file on Xcode
12. Go to Editor > Fail Transactions > select any failure except the "unknown"
13. Re-run the app
13. Go to Settings > Developer > tap "Set to No Plus"
6. Go to Podcasts > tap the Folder button
7. Buy any plan
1. ✅ An error should be displayed

### Test messages

1. Open the `.storekit` file
2. Add free trial in all 4 available tiers, choose anything you want but make sure they're different
13. Re-run the app
13. Go to Settings > Developer > tap "Set to No Plus"
1. ✅ Make sure the correct prices and trial period are shown

### Different devices

Please test the screen on smaller devices and big devices.

## Checklist
